### PR TITLE
Add cron command for DockerFile

### PR DESCRIPTION
**Pull request by GuillaumeBarreau**
## Context 

Setting Up Automatic Backups with Cron

Adding 3 new environment variables to automate the execution of backups on the server."

Environment Variables

**DAYS_TO_KEEP:**

Description: This environment variable represents the number of days to retain backup files. It is used in the cleanup process to remove backup files older than the specified duration. Example Usage: If set to 7, backup files older than 7 days will be deleted during the cleanup.

**BACKUP_CRON_EXPRESSION:**

Description: This environment variable defines the cron expression for scheduling automatic backups. It determines when the backup script should run at specific intervals. Example Usage: If set to 0 2 * * *, the backup script will run every day at 2:00 AM.

**BACKUP_ENABLED:**

Description: This environment variable indicates whether automatic backups are enabled or not. If set to a non-empty value, automatic backups will be enabled; otherwise, they will be disabled. Example Usage: If set to any non-empty string (e.g., true or 1), the backup script will be configured and executed as per the specified cron schedule.

## Test instructions

You can try to set up these variables with this value if you want to save backups every minute.

```bash
      BACKUP_ENABLED=true
      BACKUP_CRON_EXPRESSION=* * * * *
```

## Checklist before requesting a review

* [x] I have performed a self-review of my code
* [x] I've added documentation about this change to the README.
* [x] I've not introduced breaking changes.

### CODE REVIEW

**Comment by thijsvanloef** [comment]
Thanks for the PR, I will check it once I get home from work!

File: README.md [hunk 1]
```diff
@@ -162,6 +162,25 @@ docker exec palworld-server backup
...
 This will create a backup at `/palworld/backups/`
 
+## Setting Up Automatic Backups with Cron
+
+This section guides you through the process of configuring automatic backups using Cron. Use the provided command to set up the necessary Cron job for your Palworld server within a Docker environment.
+
+The configured backup system automatically deletes backups older than 7 days to manage storage efficiently.
+
+```bash
+docker exec palworld-server cron /etc/cron.d/backups-cron
```
Comment (thijsvanloef) [inline README.md:172]: This requires manually running this command through exec, I would prefer to have a enviroment variable to enable/disable the cron backup funcionality:
ENABLE_BACKUP=true/false
BACKUP_INTERVAL="0 0 * * *"

**Review by thijsvanloef — CHANGES_REQUESTED** [review]

File: Dockerfile [hunk 2]
```diff
@@ -5,12 +5,19 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
...
     && apt-get clean \
     && rm -rf /var/lib/apt/lists/*
 
 RUN wget -q https://github.com/itzg/rcon-cli/releases/download/1.6.4/rcon-cli_1.6.4_linux_amd64.tar.gz -O - | tar -xz
 RUN mv rcon-cli /usr/bin/rcon-cli
 
+RUN rm -f /var/run/crond.pid
+COPY ./scripts/backup.sh /usr/local/bin/backup
+RUN chmod +x /usr/local/bin/backup
+RUN echo "0 3 * * * root bash /usr/local/bin/backup" > /etc/cron.d/backups-cron
```
[the remark this answers is not in the document]
Comment (GuillaumeBarreau) [reply Dockerfile]: Me too :D, I wanted to incorporate environment variables in a second pull request due to a lack of time. However, I ultimately corrected this directly this morning after finding some free time.

File: README.md (hunk 1 above)
Comment (GuillaumeBarreau) [reply README.md]: Ajusted

File: Dockerfile [hunk 3]
```diff
@@ -26,7 +31,9 @@ ENV PORT= \
     RCON_ENABLED=true \
     RCON_PORT=25575 \
     QUERY_PORT=27015 \
-    TZ=UTC
```
Comment (thijsvanloef) [inline Dockerfile:29]: I believe this was a merge conflict issue, however I would like to keep this ENV :stuck_out_tongue:

File: docker-compose.yml [hunk 4]
```diff
@@ -1,6 +1,7 @@
 services:
    palworld:
-      image: thijsvanloef/palworld-server-docker:latest
+      image: test/palworld-server-docker:latest
```
Comment (thijsvanloef) [inline docker-compose.yml:3]: Leftover from your testing, please revert

File: docker-compose.yml [hunk 5]
```diff
@@ -16,7 +17,10 @@ services:
          - RCON_PORT=25575
          - TZ=UTC
          - ADMIN_PASSWORD="adminPasswordHere"
-         - COMMUNITY=false  # Enable this if you want your server to show up in the community servers tab, USE WITH SERVER_PASSWORD!
```
Comment (thijsvanloef) [inline docker-compose.yml:19]: Please do not remove the comment

**Review by thijsvanloef — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 5 above)
Comment (GuillaumeBarreau) [reply docker-compose.yml:19]: My bad

File: Dockerfile (hunk 3 above)
Comment (thijsvanloef) [reply Dockerfile:29]: line 18 in the docker-compose.yml, this is the Dockerfile :slightly_smiling_face:

File: Dockerfile [hunk 6]
```diff
@@ -5,12 +5,17 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
     xdg-user-dirs=0.17-2 \
     procps=2:3.3.17-5 \
     wget=1.21-1+deb11u1 \
+    && apt-get install -y cron \
```
Comment (thijsvanloef) [inline Dockerfile:8]: this is already a list of packages to install, please just use `cron=<version>`

File: Dockerfile [hunk 7]
```diff
@@ -26,7 +31,10 @@ ENV PORT= \
     RCON_ENABLED=true \
     RCON_PORT=25575 \
     QUERY_PORT=27015 \
-    TZ=UTC
+    TZ=UTC \
+    BACKUP_ENABLED=true \
+    DAYS_TO_KEEP=7 \
```
Comment (thijsvanloef) [inline Dockerfile:36]: Would preferably set this to "BACKUPS_TO_KEEP" and not days

File: docker-compose.yml [hunk 8]
```diff
@@ -17,6 +17,9 @@ services:
          - TZ=UTC
          - ADMIN_PASSWORD="adminPasswordHere"
          - COMMUNITY=false  # Enable this if you want your server to show up in the community servers tab, USE WITH SERVER_PASSWORD!
+         - BACKUP_ENABLED=false
+         - DAYS_TO_KEEP=7
+         - BACKUP_CRON_EXPRESSION=0 0 * * *
```
Comment (thijsvanloef) [inline docker-compose.yml:22]: This needs to be quoted

File: docker-compose.yml [hunk 9]
```diff
@@ -17,6 +17,9 @@ services:
          - TZ=UTC
          - ADMIN_PASSWORD="adminPasswordHere"
          - COMMUNITY=false  # Enable this if you want your server to show up in the community servers tab, USE WITH SERVER_PASSWORD!
+         - BACKUP_ENABLED=false
```
Comment (thijsvanloef) [inline docker-compose.yml:20]: Backups are enabled by default in the Dockerfile. Here you set it to false, what is the preferred status of this setting?

File: scripts/start.sh [hunk 10]
```diff
@@ -67,6 +67,21 @@ if [ -n "${RCON_PORT}" ]; then
     sed -i "s/RCONPort=[0-9]*/RCONPort=$RCON_PORT/" /palworld/Pal/Saved/Config/LinuxServer/PalWorldSettings.ini
 fi
 
+if [[ -n "${BACKUP_ENABLED}" ]]; then
```
Comment (thijsvanloef) [inline scripts/start.sh:70]: this needs to be if BACKUP_ENABLED = true, now it turns backups on even if the status is false

File: scripts/start.sh [hunk 11]
```diff
@@ -67,6 +67,21 @@ if [ -n "${RCON_PORT}" ]; then
...
+    echo "BACKUP_ENABLED=${BACKUP_ENABLED}"
+
+    if [[ -z "${BACKUP_CRON_EXPRESSION}" ]]; then
+        printf "\e[0;31m***** BACKUP_CRON_EXPRESSION NOT SET *****\e[0m\n"
+        exit 1
+    else
+        echo "BACKUP_CRON_EXPRESSION=${BACKUP_CRON_EXPRESSION}"
+    fi
+
+    echo "${BACKUP_CRON_EXPRESSION} root bash /usr/local/bin/backup" > /etc/cron.d/backups-cron
```
Comment (thijsvanloef) [inline scripts/start.sh:80]: this needs to be in the "else" clause, this will fail if BACKUP_CRON_EXPRESSION is empty

File: scripts/backup.sh [hunk 12]
```diff
@@ -1,8 +1,24 @@
 #!/bin/bash
 
 DATE=$(date +"%Y-%m-%d_%H-%M-%S")
-FILE_PATH="/palworld/backups/palworld-save-${DATE}.tar.gz"
+DESTINATION_PATH="/palworld/backups"
+FILE_PATH="${DESTINATION_PATH}/backup_palworld_${DATE}.tar.gz"
+
+if [ ! -f ${FILE_PATH} ]; then
+    echo "\e[0;32m***** CREATING BACKUPS FOLDER *****\e[0m\n"
```
Comment (thijsvanloef) [inline scripts/backup.sh:8]: replace echo with printf

**Review by thijsvanloef — CHANGES_REQUESTED** [review]
I've tested this on my local machine, it does not seem to work. I also found some major issues with the code

File: scripts/start.sh (hunk 11 above)
Comment (fryfrog) [inline scripts/start.sh:80]: Can you do this as the `steam` user instead?

**Pull request closed without merging**